### DISCUSSION TITLE
pda 使用问题

### DISCUSSION THREAD

**Discussion by v1xingyue** [creatorsand/solana-co-learn#7] [Q&A] [answered]
用户wallet 使用 findProgramAddress +  solana合约  id ，申请 pda 。
申请到数据空间后，然后，借由 合约 进行交互(写入)。
那是不是说，程序是可以操作 用他自己id 申请的所有pda。
或者，是不是有一种机制，创建出来的pda,只能允许某些钱包地址授权的情况下进行写入操作？ 

举个例子: 

有两个账户 A, B 
合约 C

A 使用 C 申请一个pda a , 
B 使用 C 申请一个pda b , 

现在的问题是 , A 和 B 是不是都可以通过合约C 来完成对 a , b 的写入。 
或者有一种机制，可以限制 a 只能由 A 来写入。

**Comment by DaviRain-Su** [comment, marked as answer]
- pda我的理解是根据seed 还有program id 生成的
 -  如果seed 中有用户的pubkey 那 只能这个用户调用他自己的pda地址
 - pda 换句话来说就是 一种找到数据存放的返回式 根据自己定义的seed 去查找数据存储账户

**Reply by v1xingyue** [reply]
pda account 的地址，确实是由seed 和 program id 生成的。
数据写入控制，pda 的 owner 决定了该数据只能由指定的 合约 写入。

问题实现方法: 

1.  seed 中混入入生成方 （实例中的A B 账户）的key 
2.  同时，在合约 中判断: 传入的key 必须为 signer。 

这样就决定了，我生成的数据只能由我自己控制。
